### ISSUE TITLE
Build fails with clang-5.0.1: error: 'a' is a private member of 'boost::math::quaternion<float>'

### ISSUE THREAD

**Issue by yurivict** [BALL-Project/ball#645]
```
In file included from /usr/ports/science/ball/work/ball-1_4_3-BETA1-749-g9a6dc8f82/source/STRUCTURE/structureMapper.C:10:
/usr/ports/science/ball/work/ball-1_4_3-BETA1-749-g9a6dc8f82/include/BALL/MATHS/quaternion.h:375:9: error: 'a' is a private member of 'boost::math::quaternion<float>'
                this->a = (T)1;
                      ^
/usr/ports/science/ball/work/ball-1_4_3-BETA1-749-g9a6dc8f82/include/BALL/MATHS/quaternion.h:294:9: note: in instantiation of member function 'BALL::TQuaternion<float>::setIdentity' requested here
                this->setIdentity();
                      ^
/usr/ports/science/ball/work/ball-1_4_3-BETA1-749-g9a6dc8f82/source/STRUCTURE/structureMapper.C:303:14: note: in instantiation of member function 'BALL::TQuaternion<float>::TQuaternion' requested here
                Quaternion rotation_quat;
                           ^
```

**Comment by philthiel** [comment]
We probably should switch to eigen3 or Qt quaternions in the very near future?

**Comment by tkemmer** [comment]
Yeah, I guess both `Eigen::Quaternion` and `QQuaternion` would be reasonable replacements for `BALL::TQuaternion` (or at least a more suitable base class than `boost::math::quaternion` - in case we want to keep `TQuaternion`).

On first glance, it seems that `QQuaternion` does provide the same functionality as `TQuaternion`, while eigen3 lacks at least replacements for the `{to,from}{Euler,Axis}Angle()` functions. However, this would make BALL (core) additionally depend on QtGui (which doesn't have a nice ring to it tbh).

In the very short run, we might want to consider sticking with `boost::math::quaternion`, though, and replace direct access to base class members by the respective getters and copy/move assignments as changing the base class will break some code. Even when opting for a full replacement, we should consider fixing the issue (or requiring a boost version strictly below 1.66) and deprecating `TQuaternion` first.

**Comment by philthiel** [comment]
Totally agree!

Eigen3 provides access to/from AxisAngle via Eigen::AngleAxis, which can be constructed from a quaternion. It also provides an eulerAngles() function somewhere.

**Comment by yurivict** [comment]
This holds creation of the FreeBSD port, because the build is broken.
